mobile_verifier/src/unique_connections/db.rs

```diff
@@ -0,0 +1,146 @@
+use std::ops::Range;
+
+use chrono::{DateTime, Utc};
+use file_store::unique_connections::UniqueConnectionsIngestReport;
+use futures::TryStreamExt;
+use helium_crypto::PublicKeyBinary;
+use sqlx::{FromRow, PgPool, Postgres, Transaction};
+
+use super::UniqueConnectionCounts;
+
+pub async fn get(
+    db: &PgPool,
+    reward_period: &Range<DateTime<Utc>>,
+) -> anyhow::Result<UniqueConnectionCounts> {
+    #[derive(FromRow)]
+    struct UniqueConnections {
+        hotspot_pubkey: PublicKeyBinary,
+        #[sqlx(try_from = "i64")]
+        unique_connections: u64,
+    }
+
+    let rows = sqlx::query_as::<_, UniqueConnections>(
+        r#"
+            SELECT DISTINCT ON(hotspot_pubkey, received_timestamp)
+                hotspot_pubkey, unique_connections
+            FROM unique_connections
+            WHERE received_timestamp >= $1 AND received_timestamp <= $2
```

Comment:
should this not be `received_timestamp >= $1 AND received_timestamp < $2` ? You will have entries that apply to consecutive periods otherwise

Reply:
Order by should be `ORDER BY hotspot_key, received_timestamp DESC` so we also use the last one sent within the rewardable epoch bounds

Reply:
Order by fixed in https://github.com/helium/oracles/pull/904/commits/d22c7261792628ae4312c6f4157585174711e243

exclusive end range fixed in https://github.com/helium/oracles/pull/904/commits/d22c7261792628ae4312c6f4157585174711e243